lib/migration.c

```diff
@@ -218,6 +218,61 @@ migration_feature_set(vfu_ctx_t *vfu_ctx, uint32_t flags, void *buf)
     return -EINVAL;
 }
 
+ssize_t
+handle_mig_data_read(vfu_ctx_t *vfu_ctx, vfu_msg_t *msg)
+{
+    struct migration *migr = vfu_ctx->migration;
+    struct vfio_user_mig_data_without_data *req = msg->in.iov.iov_base;
+
+    if (vfu_ctx->migration == NULL) {
+        return -EINVAL;
+    }
+
+    if (migr->state != VFIO_DEVICE_STATE_PRE_COPY
+        && migr->state != VFIO_DEVICE_STATE_STOP_COPY) {
+        return -EINVAL;
+    }
+
+    msg->out.iov.iov_len = sizeof(req) + req->size;
+    msg->out.iov.iov_base = calloc(1, msg->out.iov.iov_len);
```

Comment:
check for memory allocation failure

Reply:
Also, we shouldn't blindly allocate memory as `msg->out.iov.iov_len` might be too large causing us to crash. I think there's some maximum message parameter specified during the negotiation phase?